# CUSTOMER CHURN

## Business Understanding


In [ ]:
# Read the CSV file into a Pandas DataFrame
# Display names of columns

df = pd.read_csv('dataset\Train (2).csv')
df.columns

NAMES OF COLUMNS AND DEFINITIONS

* REGION:---------The location of each client
* TENURE:---------Duration in the network
* MONTANT:--------Top-up amount
* FREQUENCE_RECH:-A number of times the customer refilled
* REVENUE:--------Monthly income of each client
* ARPU_SEGMENT:---Income over 90 days/3
* FREQUENCE:------Number of times the client has made an income
* DATA_VOLUME:----Number of connections
* ON_NET:---------Inter expresso call
* ORANGE:---------Call to Orange
* TIGO:-----------Call to Tigo
* ZONE1:----------Call to Zones 1
* ZONE2:----------Call to Zones 2
* MRG:------------A client who is going
* REGULARITY:-----Number of times the client is active for 90 days
* TOP_PACK:-------The most active packs
* FREQ_TOP_PACK:--Number of times the client has activated the top pack packages
* CHURN:----------Variable to predict - Target


### Business Goal

To develop a predictive model that determines the likelihood for a customer to churn(to stop purchasing airtime and data from the client)

### Hypothesis

Analytical Questions

1) What is the effect of tenure on churn rate?

2) Which region is churning most?

3) Is there an indication of regularity on churn?

4) What is the effect of refill on churn?

5)

## Data Understanding

In [6]:
# use .info to get a summary(index, column data types, non-null values) of dataframe. 

df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1077024 entries, 0 to 1077023
Data columns (total 19 columns):
 #   Column          Non-Null Count    Dtype  
---  ------          --------------    -----  
 0   user_id         1077024 non-null  object 
 1   REGION          652687 non-null   object 
 2   TENURE          1077024 non-null  object 
 3   MONTANT         699139 non-null   float64
 4   FREQUENCE_RECH  699139 non-null   float64
 5   REVENUE         714669 non-null   float64
 6   ARPU_SEGMENT    714669 non-null   float64
 7   FREQUENCE       714669 non-null   float64
 8   DATA_VOLUME     547261 non-null   float64
 9   ON_NET          683850 non-null   float64
 10  ORANGE          629880 non-null   float64
 11  TIGO            432250 non-null   float64
 12  ZONE1           84898 non-null    float64
 13  ZONE2           68794 non-null    float64
 14  MRG             1077024 non-null  object 
 15  REGULARITY      1077024 non-null  int64  
 16  TOP_PACK        626129 non-null   ob

In [9]:
# count the number of missing (null or NaN) values in each column
# compute the percentage of missing values in each column to decide which columns to drop

missing_values = df.isnull().sum()

# Calculate the percentage of missing values for each column
percentage_missing = (missing_values / len(df)) * 100

# Display the percentage of missing values for each column
print(percentage_missing)

user_id            0.000000
REGION            39.399029
TENURE             0.000000
MONTANT           35.086033
FREQUENCE_RECH    35.086033
REVENUE           33.644097
ARPU_SEGMENT      33.644097
FREQUENCE         33.644097
DATA_VOLUME       49.187669
ON_NET            36.505593
ORANGE            41.516624
TIGO              59.866261
ZONE1             92.117353
ZONE2             93.612584
MRG                0.000000
REGULARITY         0.000000
TOP_PACK          41.864898
FREQ_TOP_PACK     41.864898
CHURN              0.000000
dtype: float64


The following columns will be dropped because of redundancy and percentage of missing values.  
* user_id : this column is redundant because its roles is captured by the DataFrame's index
* TIGO: It has a large percentage of missing values (59.8%)
* ZONE1: It has a large percentage of missing values (92.1%)
* ZONE2: it has a large percentage of missing values (93.6%)

In [11]:
#list columns to drop and drop them inplace

columns_to_drop = ['user_id','TIGO','ZONE1','ZONE2']
df.drop(columns=columns_to_drop, inplace=True)
df.head()

REGION        TENURE  MONTANT  FREQUENCE_RECH  REVENUE  ARPU_SEGMENT  \
0        DAKAR  K > 24 month  20000.0            47.0  21602.0        7201.0   
1          NaN  K > 24 month      NaN             NaN      NaN           NaN   
2          NaN  K > 24 month      NaN             NaN      NaN           NaN   
3  SAINT-LOUIS  K > 24 month   7900.0            19.0   7896.0        2632.0   
4        DAKAR  K > 24 month  12350.0            21.0  12351.0        4117.0   

   FREQUENCE  DATA_VOLUME  ON_NET  ORANGE MRG  REGULARITY  \
0       52.0       8835.0  3391.0   396.0  NO          62   
1        NaN          NaN     NaN     NaN  NO           3   
2        NaN          NaN     NaN     NaN  NO           1   
3       25.0       9385.0    27.0    46.0  NO          61   
4       29.0       9360.0    66.0   102.0  NO          56   

                         TOP_PACK  FREQ_TOP_PACK  CHURN  
0  On net 200F=Unlimited _call24H           30.0      0  
1                             NaN            NaN      0  
2                             NaN            NaN      0  
3                Data:490F=1GB,7d            7.0      0  
4           All-net 500F=2000F;5d           11.0      0

## Data Preparation

## Modelling 

In [1]:
import pandas as pd
import numpy as np